### PR TITLE
localproxy: init at 3.1.0

### PR DESCRIPTION
**Pull request by 70m6**
###### Description of changes

added the localproxy binary from aws
one thing to note, the repo suggests protobuf 3.17, although that seems to be deprecated by nixpkgs
I tested it using 3.19, and looked to have no issues 

This PR closes #137163

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/localproxy/default.nix
Comment (drupol) [inline pkgs/applications/networking/localproxy/default.nix:4]: I think we should use the `finalAttrs` pattern here.

Since this is your first PR, let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

**Review by drupol — CHANGES_REQUESTED** [review]
Hi ! 

Thanks for your contribution.

I made a few comments, let me know if you need some assistance.

Thanks!

**Comment by 70m6** [comment]
thanks for the clarification, I just used alejandra for the formatter, i hope that's alright

**Review by drupol — COMMENTED** [review]
This is better! Thanks.

Now, you need to sort your commits.

1. 1 commit for adding yourself to `maintainers-list.nix`
2. 1 commit for the rest

**Comment by 70m6** [comment]
okay, I hope I got it now :^)

File: maintainers/maintainer-list.nix
Comment (drupol) [inline maintainers/maintainer-list.nix:15565]: How about adding the `github` and `githubId` attributes ?

Comment (70m6) [reply maintainers/maintainer-list.nix:15565]: i'd honestly prefer being contacted by email if that's okay

**Review by drupol — APPROVED** [review]

File: maintainers/maintainer-list.nix
Comment (leona-ya) [reply maintainers/maintainer-list.nix:15565]: `githubId` and `github` is helpful as then you'll be automatically requested as a reviewer in github, when someone does a PR on one of your packages

Comment (drupol) [reply maintainers/maintainer-list.nix:15565]: @70m6 Awaiting for your feedback on this before merging the PR.

File: pkgs/applications/networking/localproxy/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/localproxy/default.nix:17]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, cmake
, openssl
, protobuf3_19
, catch2
, boost181
, icu
}:
let
  boost = pkgs.boost181.override { enableStatic = true; };
in
stdenv.mkDerivation (finalAttrs: {
  pname = "localproxy";
  version = "3.1.0";
```

File: pkgs/applications/networking/localproxy/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/localproxy/default.nix:20]:
```suggestion
```

File: pkgs/applications/networking/localproxy/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/localproxy/default.nix:30]:
```suggestion
  nativeBuildInputs = [ cmake ];

  buildInputs = [ openssl protobuf3_19 catch2 boost icu ];

  # causes redefinition of _FORTIFY_SOURCE
  hardeningDisable = [ "fortify3" ];
```

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:15565]:
```suggestion
    name = "tom barrett";
    github = "70m6";
    githubId = 105207964;
```

Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:15565]: > i'd honestly prefer being contacted by email if that's okay

You will likely only ever receive pings via GitHub with some rare exceptions.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/localproxy/default.nix
Comment (drupol) [inline pkgs/applications/networking/localproxy/default.nix:12]:
```suggestion
  boost = boost181.override { enableStatic = true; };
```

File: pkgs/applications/networking/localproxy/default.nix
Comment (drupol) [reply pkgs/applications/networking/localproxy/default.nix]: I see that this comment has been marked as resolved, but nothing has changed in your file.

Can you explain why this is needed at least?

Comment (70m6) [reply pkgs/applications/networking/localproxy/default.nix]: I thought he just suggested spaces to go between the "'s and ['s ["fortify3"], which I did.
It's needed like in many other places of nixpkgs (git grep _FORTIFY_SOURCE), without this you would get compiler errors.

Comment (drupol) [reply pkgs/applications/networking/localproxy/default.nix]: Thanks.

**Pull request merged**
